mongo/datadog_checks/mongo/mongo.py

```diff
@@ -659,207 +300,42 @@ def total_seconds(td):
                 else:
                     raise e
 
-        # If these keys exist, remove them for now as they cannot be serialized
-        try:
-            status['backgroundFlushing'].pop('last_finished')
-        except KeyError:
-            pass
-        try:
-            status.pop('localTime')
-        except KeyError:
-            pass
-
         dbnames = cli.list_database_names()
         self.gauge('mongodb.dbs', len(dbnames), tags=tags)
 
         for db_n in dbnames:
-            db_aux = cli[db_n]
-            dbstats[db_n] = {'stats': db_aux.command('dbstats')}
-
-        # Go through the metrics and save the values
-        for metric_name in self.metrics_to_collect:
-            # each metric is of the form: x.y.z with z optional
-            # and can be found at status[x][y][z]
-            value = status
-
-            if metric_name.startswith('stats'):
-                continue
-            else:
-                try:
-                    for c in metric_name.split("."):
-                        value = value[c]
-                except KeyError:
-                    continue
-
-            # value is now status[x][y][z]
-            if not isinstance(value, (int, long, float)):
-                raise TypeError(
-                    u"{0} value is a {1}, it should be an int, a float or a long instead.".format(
-                        metric_name, type(value)
-                    )
-                )
-
-            # Submit the metric
-            submit_method, metric_name_alias = self._resolve_metric(metric_name, self.metrics_to_collect)
-            submit_method(self, metric_name_alias, value, tags=tags)
-
-        for st, value in iteritems(dbstats):
-            for metric_name in self.metrics_to_collect:
-                if not metric_name.startswith('stats.'):
-                    continue
-
-                try:
-                    val = value['stats'][metric_name.split('.')[1]]
-                except KeyError:
-                    continue
-
-                # value is now status[x][y][z]
-                if not isinstance(val, (int, long, float)):
-                    raise TypeError(
-                        u"{0} value is a {1}, it should be an int, a float or a long instead.".format(
-                            metric_name, type(val)
-                        )
-                    )
-
-                # Submit the metric
-                metrics_tags = tags + [
-                    u"cluster:db:{0}".format(st),  # FIXME: 8.x, was kept for backward compatibility
-                    u"db:{0}".format(st),
-                ]
-
-                submit_method, metric_name_alias = self._resolve_metric(metric_name, self.metrics_to_collect)
-                submit_method(self, metric_name_alias, val, tags=metrics_tags)
+            collector = DbStatCollector(self, db_n, tags)
```

Comment:
(Nit) If `db_name` wouldn't collide with anything, should we go ahead and rename this, for readability?

```suggestion
            collector = DbStatCollector(self, db_name, tags)
```